motionplan/tpSpaceRRT.go

```diff
@@ -799,12 +810,22 @@ func (mp *tpSpaceRRTMotionPlanner) setupTPSpaceOptions() {
 // distances can be computed in TP space using the given PTG.
 func (mp *tpSpaceRRTMotionPlanner) make2DTPSpaceDistanceOptions(ptg tpspace.PTGSolver) *plannerOptions {
 	opts := newBasicPlannerOptions(mp.frame)
-	segMetric := func(seg *ik.Segment) float64 {
+	mu := sync.RWMutex{}
+	memo := map[[2]spatialmath.Pose]distanceAndSolution{}
+	// TODO: Refactor this to use generic memoization
+	segMetricWithSolution := func(seg *ik.Segment) distanceAndSolution {
+		memoKey := [2]spatialmath.Pose{seg.StartPosition, seg.EndPosition}
+		mu.RLock()
+		memoResult, ok := memo[memoKey]
+		mu.RUnlock()
+		if ok {
+			return memoResult
```

Comment:
Do you have a sense of how often this is hit? Might be worth sticking a print statement here or an incrementing counter and running some of the tpspace tests. It seems like it would be infrequent.